doc/lua_api.md

```diff
@@ -6881,6 +6883,33 @@ Server
       this can make transfer of bigger files painless (if set up). Nevertheless
       it is advised not to use dynamic media for big media files.
 
+IPC
+---
+
+The engine provides a generalized mechanism to enable sharing data between the
+different Lua environments (main, mapgen and async).
+
+* `minetest.ipc_get(key)`:
+  * Read a value from the shared data area.
+  * `key`: string, should use the `"modname:thing"` convention to avoid conflicts.
+  * return an arbitrary Lua value, or `nil` if this key does not exist
+* `minetest.ipc_set(key, value)`:
+  * Write a value to the shared data area.
+  * `key`: as above
+  * `value`: an arbitrary Lua value, cannot be or contain userdata.
+* `minetest.ipc_swap(key, value)`:
+  * Like `minetest.ipc_set` but returns the value that was at the same key
+    before writing the new one. This is atomic.
```

Comment:
Documentation requests:

- Mention that this is CAS and maybe link the Wikipedia page you linked
- Why/when should I use this?
- Usage example

Reply:
It's not CAS because no comparison is done.

Reply:
Indeed, I thought I could be clever and skip that. It of course doesn't work like this.